inc/class-component.php

```diff
@@ -585,14 +624,24 @@ public function jsonSerialize(): array {
 	 */
 	public function to_array(): array {
 
+		// Camel case config keys.
+		$this->set_config( $this->camel_case_keys( $this->get_config() ) );
+
+		// Camel case theme options.
+		$this->set_theme_options( $this->camel_case_keys( $this->get_theme_options() ) );
+		$this->set_theme( self::camel_case( $this->get_theme() ) );
+
 		// Add the theme name to the config as Irving core expects.
 		$this->set_config( 'theme_name', $this->get_theme() );
 		$this->set_config( 'theme_options', $this->get_theme_options() );
 
+		// Sanitize children.
+		$this->set_children( self::reset_array( $this->get_children() ) );
```

Comment:
is the interior `self::reset_array()` needed here as that seems to be what `$this->set_children()` does?

Reply:
I have it as a kind of "catch all in case something suspect happens", but I think it can be removed for now and addressed later if needed. Thanks.